arch/x86/cpuid.c

```diff
@@ -329,6 +329,18 @@ void guest_cpuid(struct vcpu *vcpu,
 		/*mask vmx to guest os */
 		*ecx &= ~CPUID_ECX_VMX;
 
+		/*no xsave support for guest if it is not enabled on host*/
+		if (!(*ecx & CPUID_ECX_OSXSAVE))
+			*ecx &= ~CPUID_ECX_XSAVE;
+
+		*ecx &= ~CPUID_ECX_OSXSAVE;
+		if (*ecx & CPUID_ECX_XSAVE) {
+			uint64_t cr4;
+			/*read guest CR4*/
+			cr4 = exec_vmread(VMX_GUEST_CR4);
```

Comment:
Donot know why we need to check guest CR4 here. This is equal to vCR4, right?

Reply:
Hi Eddie,  
it's to check guest CR4 (vCR4),  We can't return CPUID.ECX.OSXSAVE = 1 if guest did not set vCR4.OSXSAVE.